articles/how-to-setup-user-authentication-from-scratch-with-rails-6/index.md

```diff
@@ -0,0 +1,702 @@
+### How to set up User Authentication from Scratch with Rails 6
+
+User Authentication is fundamental in the security of web resources. While setting up user authentication in a `rails` program, [devise gem](https://rubygems.org/gems/devise) is a popular tool. However, at times it can be too big and complicated to customize especially when building a non-complicated application.
+
+### Goal
+
+In this tutorial, we will set up user authentication from scratch with Rails 6.
+
+### Prerequisites
+
+To follow along in this article, it is helpful to have:
+
+The following programs installed on your computer:
+
+- [Ruby](https://www.ruby-lang.org/en/)
+- [SQLite3](https://www.sqlite.org/)
+- [Node](https://nodejs.org)
+- [yarn](https://classic.yarnpkg.com/en/docs/install)
+- [Rails](https://rubygems.org/gems/rails) framework configured.
+- Basic working knowledge with Ruby programming language.
+- Basic understanding of Object Oriented Programming (OOP) paradigm.
+- Text editor installed. Preferrably [vscode](https://code.visualstudio.com/)
+
+### Overview
+
+- [Project Setup](#project-setup)
+- [Basic understanding of MVC](#basic-understanding-of-MVC)
+- [Configuring routes](#configuring-routes)
+- [Adding Controllers](#adding-controllers)
+- [Configuring views](#configuring-views)
+- [Resetting the password](#resetting-the-password)
+- [Setting up mailers](#setting-up-mailers)
+
+### Project Setup
+
+Generate a new rails application by running:
+
+```bash
+rails new auth -T
+```
+
+We are using the `-T` argument to exclude `Rails` default testing framework
+
+- Navigate to the newly created `auth` folder:
+
+```bash
+cd auth
+```
+
+In the app folder, `Rails` maintains files for the controllers, models, and views.
+
+### Basic understanding of MVC
+
+Model View Controller is a design pattern that divides related programming logic making it easier to reason about.
```

Comment:
Please provide a link to an article that explains more about the MVC architecture.